client/src/components/ShareDraftContent/ShareDraftContent.js

```diff
@@ -9,87 +9,74 @@ import { inject } from 'lib/Injector';
  */
-class ShareDraftContent extends Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      error: null,
-      isLoaded: false,
-      previewUrl: i18n._t('ShareDraftContent.LOADING', 'Loading...'),
-    };
-
-    this.handleToggle = this.handleToggle.bind(this);
-  }
-
-  /**
-   * Do nothing, workaround for un/controlled component warnings in React
-   */
-  handleInputChange() {
-    // noop
-  }
+const ShareDraftContent = ({
+  id = 'share-draft-content',
+  className,
+  button = {
+    icon: 'share',
+    title: i18n._t('ShareDraftContent.SHARE', 'Share'),
+    tooltip: i18n._t('ShareDraftContent.SHARE_DRAFT_CONTENT', 'Share draft content'),
+  },
+  popover = {
+    title: i18n._t('ShareDraftContent.SHARE_DRAFT_CONTENT', 'Share draft content'),
+  },
+  links = {
+    learnMore: '',
+  },
+  PopoverField,
+}) => {
+  const [error, setError] = useState(null);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [previewUrl, setPreviewUrl] = useState(i18n._t('ShareDraftContent.LOADING', 'Loading...'));
+  const linkRef = useRef(null);
 
   /**
-   * What to do when the "Share" button is clicked and the popover is opened. If it's already
-   * loaded then the link should be selected, otherwise it should generate (and then select)
-   * the link.
+   * Ensure the link input's contents is selected
    */
-  handleToggle() {
-    const { isLoaded } = this.state;
-    if (!isLoaded) {
-      // Generate the link
-      this.generateShareDraftLink();
-    } else {
-      // Auto select the link text if it's already been generated
-      this.selectLink();
+  const selectLink = () => {
+    if (linkRef.current) {
+      linkRef.current.select();
     }
-  }
+  };
 
   /**
    * Generate and/or get the preview draft URL from the CMS, setting it to the state once
    * completed.
    */
-  generateShareDraftLink() {
-    const { links: { generateLink } } = this.props;
-
+  const generateShareDraftLink = () => {
+    const { generateLink } = links;
     return fetch(generateLink, { credentials: 'same-origin' })
       .then(response => response.text())
-      .then(
-        (response) => {
-          this.setState({
-            isLoaded: true,
-            previewUrl: response,
-          });
-
-          // Auto-select the link text
-          this.selectLink();
-        },
-        () => {
-          this.setState({ error: true });
-        }
+      .then(responseText => {
+        setIsLoaded(true);
+        setPreviewUrl(responseText);
+        selectLink();
+      }, () => setError(true)
       );
-  }
+  };
 
   /**
-   * Ensure the link input's contents is selected
+   * What to do when the "Share" button is clicked and the popover is opened. If it's already
+   * loaded then the link should be selected, otherwise it should generate (and then select)
+   * the link.
    */
-  selectLink() {
-    if (this.linkRef) {
-      this.linkRef.select();
+  const handleToggle = () => {
```

Comment:
Moved down as is needs to be defined below generateShareDraftLink() and selectLink()